packages/webdriver-utils/src/providers/genericProvider.js

```diff
@@ -0,0 +1,127 @@
+import utils from '@percy/sdk-utils';
+import tmp from 'tmp';
+import fs from 'fs/promises';
+
+import CommonMetaDataResolver from '../metadata/commonMetaDataResolver.js';
+import Tile from '../util/tile.js';
+import Driver from '../driver.js';
+
+const log = utils.logger('webdriver-utils:genericProvider');
+// TODO: Need to pass parameter from sdk and catch in cli
+const CLIENT_INFO = 'local-poc-poa';
+const ENV_INFO = 'staging-poc-poa';
+
+export default class GenericProvider {
+  constructor(
+    sessionId,
+    commandExecutorUrl,
+    capabilities,
+    sessionCapabilites
+  ) {
+    this.sessionId = sessionId;
+    this.commandExecutorUrl = commandExecutorUrl;
+    this.capabilities = capabilities;
+    this.sessionCapabilites = sessionCapabilites;
+    this.driver = null;
+    this.commonMetaData = null;
+    this.debugUrl = null;
+  }
+
+  async createDriver() {
+    this.driver = new Driver(this.sessionId, this.commandExecutorUrl);
+    const caps = await this.driver.getCapabilites();
+    this.commonMetaData = await CommonMetaDataResolver.resolve(this.driver, caps, this.capabilities);
+  }
+
+  static supports(_commandExecutorUrl) {
+    return true;
+  }
+
+  async screenshot(name) {
+    let fullscreen = false;
+
+    const tag = await this.getTag();
+    const tiles = await this.getTiles(fullscreen);
+    await this.setDebugUrl();
+
+    log.debug(`${name} : Tag ${JSON.stringify(tag)}`);
+    log.debug(`${name} : Tiles ${JSON.stringify(tiles)}`);
+    log.debug(`${name} : Debug url ${this.debugUrl}`);
+    return await utils.postComparison({
```

Comment:
dont use `postComparison` if possible and call function directly. Its better performance + ram usage. [ No need to serialize and deserialize everything again in same process ]